### ISSUE TITLE
Create Java 9 modules

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#4085]
Java 9 modules solve many causes of version collision issues, which would greatly help users. Granted, it's not helpful unless people are actually on Java 9.

This is probably blocked by #3522. It'd probably be reasonable to wait for #4084 as well.

https://guides.gradle.org/building-java-9-modules/

**Comment by hohwille** [comment]
```
[WARNING] ***********************************************************************************************************************************************
[WARNING] * Required filename-based automodules detected: [protobuf-java-3.9.2.jar]. Please don't publish this project to a public artifact repository! *
[WARNING] ***********************************************************************************************************************************************
```

You can easily add a `MANIFEST.MF` with module name. Via Maven I have provided this for TeaVM here:
https://github.com/konsoletyper/teavm/pull/490/files

**Comment by ejona86** [comment]
@hohwille, `protobuf-java` is part of https://github.com/protocolbuffers/protobuf

**Comment by sanjaypujare** [comment]
Can this be considered fixed with automatic module name support?

**Comment by ejona86** [comment]
Yeah, automatic module name works. Fixed by #10413
